plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/S3FileSystemExchangeStorage.java

```diff
@@ -574,17 +619,17 @@ private void fillBuffer()
                     }
                 }
 
-                String key = keyFromUri(currentFile.getFileUri());
-                String bucketName = getBucketName(currentFile.getFileUri());
+                URI fileUri = currentFile.getFileUri();
+                String key = keyFromUri(fileUri);
+                String bucketName = getBucketName(fileUri);
                 Optional<SecretKey> secretKey = currentFile.getSecretKey();
                 for (int i = 0; i < readableParts && fileOffset < fileSize; ++i) {
                     int length = (int) min(partSize, fileSize - fileOffset);
-                    int partNumber = (int) (fileOffset / partSize + 1);
 
                     GetObjectRequest.Builder getObjectRequestBuilder = GetObjectRequest.builder()
                             .key(key)
                             .bucket(bucketName)
-                            .partNumber(partNumber);
+                            .range("bytes=" + fileOffset + "-" + (fileOffset + length - 1));
```

Comment:
Is that still OK for AWS (e.g. doesn't it impose some performance degradation)?

Reply:
<img width="1166" alt="image" src="https://user-images.githubusercontent.com/9404831/169119789-2112466a-1863-44c5-b7dd-776c43b71104.png">
To my understanding, they are equivalent. Previously we decided to use `partNumber` was to avoid accidentally reading across part boundaries. However, given that GCP is not compatible with partNumber, and community also sees incompatibilities (e.g. Netapp Storage Grid), I guess we may simply use byte range instead.